shell/platform/darwin/ios/framework/Source/FlutterKeyboardManagerTest.mm

```diff
@@ -17,364 +15,253 @@
 
-  FlutterViewController* viewController =
-      [[FlutterViewController alloc] initWithEngine:self.mockEngine nibName:nil bundle:nil];
+  id mockEngine = OCMClassMock([FlutterEngine class]);
+  FlutterViewController* viewController = [[FlutterViewController alloc] initWithEngine:mockEngine
+                                                                                nibName:nil
+                                                                                 bundle:nil];
   FlutterViewController* owner = OCMPartialMock(viewController);
   OCMStub([owner nextResponder]).andReturn(nextResponder);
-  return owner;
-}
 
-// Verify that the nextResponder returned from mockOwnerWithPressesBeginOnlyNext()
-// throws exception when pressesEnded is called.
-- (bool)testNextResponderShouldThrowOnPressesEnded API_AVAILABLE(ios(13.4)) {
-  FlutterViewController* owner = [self mockOwnerWithPressesBeginOnlyNext];
-  @try {
-    [owner.nextResponder pressesEnded:[NSSet init] withEvent:[[UIPressesEvent alloc] init]];
-    return false;
-  } @catch (...) {
-    return true;
-  }
+  XCTAssertThrowsSpecificNamed([owner.nextResponder pressesEnded:[[NSSet alloc] init]
+                                                       withEvent:[[UIPressesEvent alloc] init]],
+                               NSException, NSInternalInconsistencyException);
+
+  [mockEngine stopMocking];
 }
 
-- (void)testSinglePrimaryResponder API_AVAILABLE(ios(13.4)) {
+- (void)testSinglePrimaryResponder {
+  const UIKeyboardHIDUsage keyId = (UIKeyboardHIDUsage)0x50;
+
   FlutterKeyboardManager* manager = [[FlutterKeyboardManager alloc] init];
-  __block BOOL primaryResponse = FALSE;
-  __block int callbackCount = 0;
-  [manager addPrimaryResponder:[self mockPrimaryResponder:^(FlutterUIPressProxy* press,
-                                                            FlutterAsyncKeyCallback callback) {
-             callbackCount++;
-             callback(primaryResponse);
-           }]];
-  constexpr UIKeyboardHIDUsage keyId = (UIKeyboardHIDUsage)0x50;
-  // Case: The responder reports TRUE
-  __block bool completeHandled = true;
-  primaryResponse = TRUE;
+  id mockPrimaryResponder = OCMStrictProtocolMock(@protocol(FlutterKeyPrimaryResponder));
+  [manager addPrimaryResponder:mockPrimaryResponder];
+
+  // Case: The responder reports that is has handled the event (callback returns YES), and the
+  // manager should NOT call next action.
+  [[mockPrimaryResponder expect] handlePress:OCMOCK_ANY
+                                    callback:[OCMArg invokeBlockWithArgs:@YES, nil]];
+
+  XCTestExpectation* expectationPrimaryHandled =
+      [self expectationWithDescription:@"primary responder handled"];
+  expectationPrimaryHandled.inverted = YES;  // Fail if the manager "next action" is called.
   [manager handlePress:keyDownEvent(keyId)
             nextAction:^() {
-              completeHandled = false;
+              [expectationPrimaryHandled fulfill];
+              XCTFail();
```

Comment:
These `XCTFail`s aren't strictly necessary with the inverted expectation, but I think they make it clearer that this block shouldn't be called.

Reply:
I reverted this.